src/sentry/replays/query.py

```diff
@@ -55,7 +55,7 @@ def query_replays_collection(
     """Query aggregated replay collection."""
     conditions = []
     if environment:
-        conditions.append(Condition(Column("environment"), Op.IN, environment))
+        conditions.append(Condition(Column("agg_environment"), Op.IN, environment))
```

Comment:
out of curiousity how does this solve for the deleted case?

Reply:
ah i see -- isArchived doesn't have an environment?

Reply:
The condition is applied in the having clause.  The archive record is inserted as a null environment.  Putting it in the where clause screens out the archive record.  Whereas the having clause includes the archive record.

Reply:
Maybe more clearly.  The agg_envinroment is an aggregated column so the null environment on the archive record is ignored.  If you filter by agg_environment you can return the archived record.  If you filter by environment you do not return the archived record.

Reply:
Oh I didn't see your second comment.  You were spot on.

Reply:
good to note. 💯 catch on all of this, great stuff!!!